contracts/src/assertionStakingPool/AssertionStakingPool.sol

```diff
@@ -7,97 +7,53 @@ pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "../rollup/IRollupLogic.sol";
-import "../rollup/IRollupCore.sol";
-import "./StakingPoolErrors.sol";
+import "./AbsBoldStakingPool.sol";
+import "./interfaces/IAssertionStakingPool.sol";
 
 /// @notice Staking pool contract for target assertion.
 /// Allows users to deposit stake, create assertion once required stake amount is reached,
 /// and reclaim their stake when and if the assertion is confirmed.
-contract AssertionStakingPool {
+contract AssertionStakingPool is AbsBoldStakingPool, IAssertionStakingPool {
```

Comment:
We should also state that token should not be sent directly to this contract or else they would not be recoverable

Reply:
either that or we track the total deposit and withdrawal and allow an owner to draw excess